lib/gooddata/lcm/actions/collect_meta.rb

```diff
@@ -13,30 +13,43 @@ class CollectMeta < BaseAction
       with objects inside dashboards (reports, metrics ...) from development projects"
 
       PARAMS = define_params(self) do
-        description 'Tag Name'
+        description 'Production Tag Name'
         param :production_tag, instance_of(Type::StringType), required: false
 
         description 'Development Client Used for Connecting to GD'
         param :development_client, instance_of(Type::GdClientType), required: true
 
         description 'Synchronization Info'
         param :synchronize, array_of(instance_of(Type::SynchronizationInfoType)), required: true, generated: true
+
+        description 'Segments to search for segment-specific production tags'
+        param :segments, array_of(instance_of(Type::SegmentType)), required: false
       end
 
       class << self
         def call(params)
           results = []
 
           development_client = params.development_client
+          segments_to_tags = Helpers.segment_production_tags(params.segments)
 
           synchronize = params.synchronize.pmap do |info|
             from = info.from
             from_project = development_client.projects(from) || fail("Invalid 'from' project specified - '#{from}'")
 
-            if params.production_tag
-              objects = GoodData::Dashboard.find_by_tag(params.production_tag, project: from_project, client: development_client)
+            segment_tags = segments_to_tags[info.segment]
+            production_tags = Helpers.parse_production_tags(params.production_tag, segment_tags)
```

Comment:
I've chosen this way (merging global and segment-specific tags) because it honors the DRY principle. E.g. I have tags `basic` for all customers and `advanced` for the premium segment. By specifying `basic` globally and `advanced` in the segment, I don't have to repeat the `basic` tag in the segment scope.

I'll be happy to change the implementation if it turns out to be impractical but IMHO it's better. What do you think @korczis, @zsvoboda?